### PR TITLE
Feature/upgrade to angular 19

### PR DESCRIPTION
**Pull request by laurentgrangier**
https://github.com/eclipsesource/jsonforms/issues/2386

### CODE REVIEW

**Review by lucas-koehler — CHANGES_REQUESTED** [review]
Hi @laurentgrangier , thanks for this PR :heart: 
Many of the changes already look quite good. There are some minor comments inline.

One bigger issue seems to be the coverage reporting for the `angular-material` package. Executing `pnpm run test-cov` in the `angular-material` package no longer creates an lcov.info file. This is needed for coverage reports in PRs. You can see the expected result by executing tests on the main branch. For the `angular` package this still works.
My assumption is that there was some config in the deleted test-runner.js that configured this. Please have a look.

File: packages/angular-material/angular.json [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+          "style": "scss"
+        }
+      },
+      "root": "example",
+      "sourceRoot": "example",
+      "prefix": "app",
+      "architect": {
+        "build": {
+          "builder": "@angular-devkit/build-angular:browser",
+          "options": {
```
[the remark this answers is not in the document]
Comment (laurentgrangier) [reply packages/angular-material/angular.json:19]: I added the options again. Thank you.

File: packages/angular-material/angular.json [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+              "buildTarget": "example:build:production"
+            },
+            "development": {
+              "buildTarget": "example:build:development"
+            }
+          },
+          "defaultConfiguration": "development"
+        }
+      }
+    },
```
[the remark this answers is not in the document]
Comment (laurentgrangier) [reply packages/angular-material/angular.json]: You are right. I just copied them from the default generate file from Angular. I removed them.

File: packages/angular-material/angular.json [hunk 3]
```diff
@@ -0,0 +1,87 @@
+{
+  "$schema": "./node_modules/@angular/cli/lib/config/schema.json",
```
[the remark this answers is not in the document]
Comment (laurentgrangier) [reply packages/angular-material/angular.json]: Done.

File: packages/angular-material/package.json [hunk 4]
```diff
@@ -60,14 +60,8 @@
     ]
   },
   "peerDependencies": {
```
[the remark this answers is not in the document]
Comment (laurentgrangier) [reply packages/angular-material/package.json:62]: I don't know. I don't remember removing them. But we can revert this remove. Thank you.

**Comment by laurentgrangier** [comment]
> Hi @laurentgrangier , thanks for this PR ❤️ Many of the changes already look quite good. There are some minor comments inline.
> 
> One bigger issue seems to be the coverage reporting for the `angular-material` package. Executing `pnpm run test-cov` in the `angular-material` package no longer creates an lcov.info file. This is needed for coverage reports in PRs. You can see the expected result by executing tests on the main branch. For the `angular` package this still works. My assumption is that there was some config in the deleted test-runner.js that configured this. Please have a look.

The karma.config.js was missing in angular.json. Adding it properly should fix the missing lcov file. Can you please check? I'm not sure where I should see them in the build.

**Review by lucas-koehler — APPROVED** [review]
Hi @laurentgrangier ,
thanks for the updates. This LGTM now. Thanks for all your work :heart: 

I tested this locally with Angular 18 and 19 in the angular seed. Both worked fine :)

Regarding the code coverage: Executing the tests locally I could see that the code coverage files are now generated at expected location `packages/angular-materila/coverage/lcov.info`.
Apparently, there is another issue with the coverage report not being displayed in the PR as this is also the case for another, independent PR. Thus, nothing additional needs to be done here.

Note that I will close this PR and open a followup  one because we generally do not accept re-generated lock file contributions from third party contributors for security reasons. I could not push a re-generated one on your branch as this is disabled in your fork's settings. Thus, I'll push this to a branch here and open a new PR from there.

**Comment by lucas-koehler** [comment]
Superseded by https://github.com/eclipsesource/jsonforms/pull/2403 . Thanks again @laurentgrangier  for this great PR :)

**Comment by laurentgrangier** [comment]
Thank you for the review and the new PR. 👍

**Pull request closed without merging**